app/apollo/resolvers/common.js

```diff
@@ -20,13 +20,37 @@ const whoIs = me => {
   if (me === null || me === undefined) return 'null';
   if (me.email) return me.email;
   if (me.identifier) return me.identifier;
+  if (me.type) return me.type;
   return me._id;
 };
 
 // Validate is user is authorized for the requested action.
 // Throw exception if not.
 const validAuth = async (me, org_id, action, type, queryName, context) => {
   const {req_id, models, logger} = context;
+
+  // razeedash users (x-api-key)
+  if(me && me.type == 'userToken'){
+    const result = await models.User.userTokenIsAuthorized(me, org_id, action, type, null, context);
+    if(!result){
+      throw new ForbiddenError(
+        `You are not allowed to ${action} on ${type} under organization ${org_id} for the query ${queryName}. (using userToken)`,
+      );
+    }
+    return;
+  }
+
+  // Users that pass in razee-org-key.  ex: ClusterSubscription or curl requests
+  if(me && me.type == 'cluster'){
```

Comment:
For cluster-facing APIs, could we create `validClusterAuth` function?  I am a little worried about users could bypass User IAM enforcement by using `org-Key` to invoke other APIs.